snuba/request/schema.py

```diff
@@ -133,6 +133,9 @@ def generate_template(self) -> Any:
             "dry_run": {"type": "boolean", "default": False},
             # Flags if this a legacy query that was automatically generated by the SnQL SDK
             "legacy": {"type": "boolean", "default": False},
+            # Team and feature are used for resource attribution
+            "team": {"type": "string", "default": "<unknown>"},
```

Comment:
Food for thoughts. 
I think we will have to decide how we allocate capacity to teams. There should be two ways:
- A team asks for a quota. We assign that, we give it an id like an app key (which is what we would see here) and somewhere else (out of the snuba query path) we keep track of a data model that assigns quota ids to teams/features
- We identify a quota with the feature-team pair. This univocally assign a quota to a team directly. Though it becomes impractical during reorgs as teams can merge and split for reasons that are not technical. When that happens we would have to adjust the allocation and wreck our data.
- We identify a qutoa with a product-feature pair. That gives the same level of granularity as the team-feature pair but it is agnostic to team reorganizations. The mapping between product and team happens in a data model outside of snuba. The data consumption we store does not change.

I think the first and third are not substantially different. The third maybe can make some analysis quicker to do. The second may become hard to manage.

Reply:
I agree. I think ultimately using an API key makes more sense because it's easier to automate and lets teams use their quota how they want, instead of being tied to a specific product/feature name. I think this first step will let us get an MVP of how things are being used and we can iterate on the api key concept and how quotas are defined.